typescript/web/src/components/labeling-tool/options-tool-bar/__tests__/edit-label-class-menu.tsx

```diff
@@ -1,167 +1,83 @@
-  const original = jest.requireActual(
-    "../../../../connectors/apollo-client/schema-client"
-  );
-  return {
-    client: {
-      ...original.client,
-      clearStore: original.client.clearStore, // This needs to be passed like this otherwise the resulting object does not have the clearStore method
-      mutate: jest.fn(() => {
-        return { data: { createLabelClass: { id: "label class id" } } };
-      }),
-      mutateOriginal: original.client.mutate,
-      query: jest.fn(() => {
-        return {
-          data: {
-            labelClasses: [
-              {
-                id: "existing label class id",
-                name: "existing label class",
-                color: "0xaa45f7",
-              },
-            ],
-            label: {
-              id: "my label id",
-              labelClass: {
-                id: "previous label class id",
-              },
-            },
-          },
-        };
-      }),
-    },
-  };
-});
-
-const createDataset = async (
-  name: string,
-  datasetId: string = testDatasetId
-) => {
-  // @ts-ignore
-  return client.mutateOriginal({
-    mutation: CREATE_TEST_DATASET_MUTATION,
-    variables: {
-      name,
-      datasetId,
-      workspaceSlug: "local",
-    },
-    fetchPolicy: "no-cache",
-  });
-};
-
-const renderEditLabelClassMenu = () => {
+const renderEditLabelClassMenu = (mockLink: WildcardMockLink) => {
   return render(<EditLabelClassMenu />, {
-    wrapper: ({ children }) => (
-      <ApolloProvider client={client}>{children}</ApolloProvider>
-    ),
+    wrapper: getApolloMockWrapper(mockLink),
   });
 };
 
-beforeEach(async () => {
-  useLabelingStore.setState({
-    selectedLabelId: "my label id",
-    selectedTool: Tools.SELECTION,
-  });
-
-  await createDataset("Test dataset");
-
-  // @ts-ignore
-  await client.mutateOriginal({
-    mutation: CREATE_LABEL_CLASS_MUTATION,
-    variables: {
-      id: "existing label class id",
-      name: "existing label class",
-      color: "0xaa45f7",
-      datasetId: testDatasetId,
-    },
-  });
+beforeEach(() => {
+  act(() =>
+    useLabelingStore.setState({
+      selectedLabelId: BASIC_LABEL_DATA.id,
+      selectedTool: Tools.SELECTION,
+    })
+  );
 
   jest.clearAllMocks();
 });
 
 it("should create a class", async () => {
-  renderEditLabelClassMenu();
-
-  userEvent.type(screen.getByPlaceholderText(/Search/), "newClass{enter}");
-
-  await waitFor(() => {
-    expect(client.mutate).toHaveBeenCalledWith(
-      expect.objectContaining({
-        variables: {
-          data: expect.objectContaining({
-            name: "newClass",
-          }),
-        },
-      })
-    );
-  });
-
-  // The id is generated by the effect. We retrieve it from the first mutation
-  // which is "createLabelClass"
-  const { id } = (client.mutate as jest.Mock).mock.calls[0][0].variables.data;
+  const mockLink = getApolloMockLink(APOLLO_MOCKS);
+  renderEditLabelClassMenu(mockLink);
+  act(() =>
```

Comment:
Shouldn't you `await` the `act`?

Reply:
only when the act wraps a promise returning function :)